src/monattack.cpp

```diff
@@ -5121,6 +5165,151 @@ bool mattack::bio_op_takedown( monster *z )
     return true;
 }
 
+bool mattack::bio_op_impale( monster *z )
+{
+    if( !z->can_act() ) {
+        return false;
+    }
+
+    Creature *target = z->attack_target();
+    if( target == nullptr ||
+        !is_adjacent( z, target, false ) ||
+        !z->sees( *target ) ) {
+        return false;
+    }
+
+    bool seen = g->u.sees( *z );
+    player *foe = dynamic_cast< player * >( target );
+    if( seen ) {
+        add_msg( _( "The %1$s mechanically lunges at %2$s!" ), z->name(),
+            target->disp_name() );
+    }
+    z->moves -= 100;
+
+    if( target->uncanny_dodge() ) {
+        return true;
+    }
+
+    // Can we dodge the attack? Uses player dodge function % chance (melee.cpp)
+    if( dodge_check( z, target ) ) {
+        target->add_msg_player_or_npc( _( "You dodge it!" ),
+            _( "<npcname> dodges it!" ) );
+        target->on_dodge( z, z->type->melee_skill * 2 );
+        return true;
+    }
+
+    int dam = rng( 8, 24 );
+    if( foe == nullptr ) {
+        // Handle mons earlier - less to check for
+        target->deal_damage( z, bp_torso, damage_instance( DT_STAB, dam ) );
+        if( seen ) {
+            add_msg( _( "%1$s impales %2$s!" ), z->name(), target->disp_name() );
+        }
+        target->check_dead_state();
+        return true;
```

Comment:
So, no bleed effects for other monsters?

Reply:
Technically, no. The bleed only occurs on a "crit", so monsters will never bleed from a normal impale.

Reply:
Technically no since the bleed only triggers on a "crit" and monster never get crit by the impale.